waiter/src/waiter/metrics_sync.clj

```diff
@@ -418,3 +491,64 @@
                    router->service-id->metrics))
     (catch Exception e
       (log/error e "error in obtaining router-id->metrics data for" service-id))))
+
+(defn handle-external-metrics-request
+  "Handle incoming external instance metrics and update metrics stored in memory. Expect the json body to be in the format
+  {'cluster': 'waiter-cluster-name'
+   'service-metrics': service-id->instance-id->metric Map}
+
+   where 'service-metrics' is map service-id->instance-id->metric where the instance's metric is:
+  {'updated-at' ISO-8601 timestamp
+   'metric': {'active-request-count' non-negative-int
+              'last-request-time' ISO-8601 timestamp}}
+
+   Extra fields in the instance's metric are kept, but are not expected to be there for any component in Waiter.
+   'updated-at' timestamp is used to determine which competing metrics is more up to date. Waiter keeps only the latest.
+   'active-request-count' is used for auto-scaling and calculating total outstanding requests for a service.
+   'last-request-time' is used for Garbage Collecting and starting new versions of a service."
+  [router-metrics-agent query-state-fn {:keys [request-method] :as request}]
+  (when (not= request-method :post)
+    (throw (ex-info "Invalid request method. Only POST is supported." {:log-level :info
+                                                                       :request-method request-method
+                                                                       :status http-400-bad-request})))
+  (let [{:strs [cluster service-metrics]} (-> request ru/json-request :body)
+        throw-error-response-if-invalid-fn
+        (fn throw-error-response-if-invalid-fn
+          [valid?-fn map keys error-msg]
+          (let [val (get-in map keys)]
+            (when-not (valid?-fn val)
+              (throw (ex-info
+                       (str "Invalid '" (str/join "." keys) "' field. " error-msg)
+                       {:log-level :info
+                        :status http-400-bad-request})))))
+
+        invalid-time-error-msg "Must be ISO-8601 time."]
+
+    ; only process metrics if the cluster matches the current cluster
+    (when (not= cluster (config/retrieve-cluster-name))
```

Comment:
This is a new function, feels like we can avoid the dependency on `config/retrieve-cluster-name` and instead pass in the cluster name explicitly as a function argument.